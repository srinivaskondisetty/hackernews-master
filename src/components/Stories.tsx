import React, { Component } from 'react'
import StoryService from '../services/storyService'
import Story from './story';

interface IStoryListProps {
}

interface IStoryListState {
    storyList: number[];
}

export default class Stories extends Component<IStoryListProps, IStoryListState> {
    private storyService: StoryService;

    constructor(props: IStoryListProps) {
        super(props);
        this.storyService = new StoryService();
        this.state = {
            storyList: []
        }
    }

    async componentDidMount() {
        const storyList = (await this.storyService.getStories()).slice(0, 10);
        this.setState({ storyList });
    }

    render() {
        const stories = this.state.storyList;
        return (
            <div id="storiesContainer" className="p-10 grid grid-cols gap-5 bg-blue-100">
                <h1 className="text-center font-black text-green-700">Story Board</h1>
                {stories &&
                    stories.map((storyId) => {
                        return <Story key={storyId} storyId={storyId}></Story>
                    })
                }
                {stories.length === 0 && <b>Loading..</b>}
            </div>
        )
    }
}
