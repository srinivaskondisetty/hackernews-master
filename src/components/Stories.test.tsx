import * as React from "react";
import { shallow } from "enzyme";
import Stories from "./stories";

describe("Stories Test Cases", () => {
  test("Checking atleast one element is loaded", () => {
    const warpper = shallow(<Stories />);
    const element = warpper.find("h1");
    expect(element).toBeDefined();
    expect(element.text()).toBe("Story Board");
  });

  test("Check if stories container is displayed", () => {
    const warpper = shallow(<Stories />);
    const element = warpper.find("#storiesContainer");
    expect(element.length).toBe(1);
  });
});
