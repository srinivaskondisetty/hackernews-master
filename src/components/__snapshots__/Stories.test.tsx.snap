// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel changes the text after click 1`] = `
<div
  className="p-10 grid grid-cols gap-5 bg-blue-100"
>
  <h1
    className="text-center font-black text-green-700"
  >
    Story Board
  </h1>
  <b>
    Loading..
  </b>
</div>
`;

exports[`Stories Test Cases CheckboxWithLabel changes the text after click 1`] = `
<div
  className="p-10 grid grid-cols gap-5 bg-blue-100"
>
  <h1
    className="text-center font-black text-green-700"
  >
    Story Board
  </h1>
  <b>
    Loading..
  </b>
</div>
`;
